staging/vhost-device-gpu/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 Red Hat Inc
+//
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+pub mod device;
+pub mod protocol;
+pub mod virtio_gpu;
+
+use std::path::PathBuf;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum GpuMode {
+    ModeVirglRenderer,
+    ModeGfxstream,
+}
+
+#[derive(Debug, Clone)]
+/// This structure is the public API through which an external program
+/// is allowed to configure the backend.
+pub struct GpuConfig {
+    /// vhost-user Unix domain socket
+    socket_path: PathBuf,
+    renderer: GpuMode,
+}
+
+impl GpuConfig {
+    /// Create a new instance of the GpuConfig struct, containing the
+    /// parameters to be fed into the gpu-backend server.
+    pub const fn new(socket_path: PathBuf, renderer: GpuMode) -> Self {
+        Self {
+            socket_path,
+            renderer,
+        }
+    }
+
+    /// Return the path of the unix domain socket which is listening to
+    /// requests from the guest.
+    pub fn get_socket_path(&self) -> PathBuf {
+        PathBuf::from(&self.socket_path.as_path())
```

Comment:
Any reason why you are re-creatting a PathBuf insead of returning a ref?

Reply:
returning a ref to `pathbuf` causes borrowing issues. I can return a `String` instead to avoid this.

Reply:
@dorindabassey, we can do this here:
```Rust
pub fn get_socket_path(&self) -> &Path {
        &self.socket_path
}
```

And then `get_socket_path().to_owned()` in a caller if you actually need a copy.

Reply:
(`Path::to_path_buf`/`.socket_path().to_path_buf()`)

Reply:
Yep, my recommendation is to always return a ref and let the caller clone if needed instead of always cloning.